modules/createHistory.js

```diff
@@ -120,7 +120,7 @@ function createHistory(options={}) {
           const prevPath = createPath(location)
           const nextPath = createPath(nextLocation)
 
-          if (nextPath === prevPath)
+          if (nextPath === prevPath && deepEqual(location.state, nextLocation.state))
```

Comment:
We should probably make this check w/something that shares code w/`locationsAreEqual`, possibly prior to calling `createPath`.